src/shvec.c

```diff
@@ -36,18 +37,39 @@ int shvec_create(){
 }
 
 /*
-* Appends a value to the shvec, reallocates the array if necessary
+ * Appends a value to the shvec, reallocates the array if necessary
  */
 int shvec_append(int id, int value){
     if(shvec_array[id].size < shvec_array[id].max_size){
         shvec_array[id].size++;
         shvec_array[id].data[shvec_array[id].size] = value;
         return 0;
     } else {
+        int* old_data_ptr = shvec_array[id].data;
+        int old_size = shvec_array[id].size;
+
         // realloc and update max_size
-        // todo (Matt)
-        fprintf(stderr, "ERROR: could not resize array\n");
-        return 1;
+        while (shvec_array[id].size < shvec_array[id].max_size) {
+            shvec_array[id].max_size = shvec_array[id].max_size << 1;
+            
+            if (shvec_array[id].max_size >= SHVEC_SIZE_CAP) {
```

Comment:
Will this actually catch an overflow? if you overflow, max_size will become negative...

Reply:
oh, I see, SHVEC_SIZE_CAP is one bit smaller than the largest int